### PR TITLE
Regression test the gradle/gradle build

### PR DESCRIPTION
**Pull request by eskatos**
This PR introduces two performance tests running the `gradle/gradle` build.

The first one, in `GradleInceptionPerformanceTest`, compares the current Gradle and a baseline Gradle version running the `gradle/gradle` build at the current commit. In other words, it asserts that changes in Gradle don't introduce regressions for running the `gradle/gradle` build.

The second one, in `GradleBuildPerformanceTest`, compares the current commit and a baseline commit of `gradle/gradle` running the current Gradle. In other words, it asserts that changes in the `gradle/gradle` build don't introduce regressions.

Both tests make use of the `help` task to measure configuration time. Both also do clone the `gradle/gradle` repository from the current working copy, preventing expensive remote git cloning.

This work is being done as part of dog-fooding the Gradle Kotlin DSL on `gradle/gradle`, in order to measure the performance impact.

### CODE REVIEW

**Comment by eskatos** [comment]
[Branch Accept CI Build](https://builds.gradle.org/viewLog.html?buildId=10220483&) ✅

**Comment by oehme** [comment]
This will likely fail each time we add a new subproject, how can we mitigate that?

**Comment by eskatos** [comment]
@oehme, right, adding any new feature to the build that will have an impact on performance (e.g. a new subproject with significant configuration time) will require rebaselining. I'm not sure how we can mitigate that, or if we should.

**Comment by oehme** [comment]
Well, it adds another step to creating a new subproject (which is already too complicated imho). We should at least make sure that it can be done in one go (i.e. make the change and rebaseline at the same time as opposed to having to wait for a CI build) and add it to the "adding a new subproject" document.

**Comment by eskatos** [comment]
It's a chicken and egg problem. You need the change in in order to get the new baseline. Just like with any regression that requires rebaselining. So, same problem as with other performance tests we temporarily ignore until the new baseline is available.

It's also possible that adding a new subproject doesn't incur enough configuration time and doesn't require rebaselining. Well, typical subprojects will, but that doesn't happen that often.

I agree that it adds some burden when making changes to the build. But for now we don't assert that the `gradle/gradle` build doesn't regress. I'd say it's the price to pay.

I don't see how we can solve this. I'd be happy to learn how to break that circle! Any idea?

File: subprojects/build-scan-performance/build-scan-performance.gradle [hunk 1]
```diff
@@ -21,6 +21,9 @@ performanceReport {
 }
 
 dependencies {
+    // so that all Gradle features are available
+    performanceTestRuntime rootProject.configurations.testRuntime.allDependencies
```
Comment (bamboo) [inline subprojects/build-scan-performance/build-scan-performance.gradle]: :o: Extract `rootProject.configuration.testRuntime.allDependencies` as something like`allTestRuntimeDependencies` and reuse?

**Comment by oehme** [comment]
Well, since your baseline is a commit, you could add another commit right away and have a proper baseline that way. You'd just need to be careful when rebasing.

**Comment by eskatos** [comment]
Right! If you want to do it in one go you must be careful if rebasing/squashing while working on a branch but also do not use the `Rebase and merge` nor `Squash and merge` Github buttons when merging.

I'll add some words in this regard on the test that would fail and at the place where the baseline commit is defined. So that if one sees it fail he easily gets the process.
Would that satisfy your concerns?

**Comment by eskatos** [comment]
[Branch Accept CI Build](https://builds.gradle.org/viewLog.html?buildId=10242919) ✅

**Comment by jjohannes** [comment]
Test slack integration

**Pull request merged**